### PR TITLE
feat: Show password on login screen

### LINKED ISSUE #506: Chatwoot Mobile App - Release v2
# Goals

- Better gesture interactions and navigations.
- A fairly **consistent 60 FPS** across the supported devices
- Improves startup time  (Android)
- Consider moving to the new Architecture  [https://reactnative.dev/blog/2022/03/15/an-update-on-the-new-architecture-rollout](https://reactnative.dev/blog/2022/03/15/an-update-on-the-new-architecture-rollout)
- Verify your app is working as expected on a device (or a simulator/emulator) from the user perspective (Consider using [https://github.com/wix/detox/](https://github.com/wix/detox/), [https://www.netguru.com/codestories/automated-end-to-end-testing-in-react-native-with-detox](https://www.netguru.com/codestories/automated-end-to-end-testing-in-react-native-with-detox))

# Stability Issues

- Not reliable enough to use on low bandwidth connection. Should show an update icon whenever we are fetching data?
    - Avoid API calls on every navigation
    - Duplicate calls, block loaders, and the app feel slow.
    - The mobile app should be taking data from a local store and fetching the data in the background.
- 300ms Click.
- Less performant tabs. https://github.com/chatwoot/chatwoot-mobile-app/issues/382, https://github.com/chatwoot/chatwoot-mobile-app/issues/421, https://github.com/chatwoot/chatwoot-mobile-app/issues/397
- The sequence of API calls / local store calls.
- Online status - If the app is available in the background, update the online status regularly- https://github.com/chatwoot/chatwoot-mobile-app/issues/503
- Navigation is not smooth - https://github.com/chatwoot/chatwoot-mobile-app/issues/415
- Optimize the Flat list [https://medium.com/@giomodiogo/how-did-i-optimize-my-react-native-flatlist-edbc54acc231](https://medium.com/@giomodiogo/how-did-i-optimize-my-react-native-flatlist-edbc54acc231)

A lot of the stability issues are due to improper state management. 

- Every action that triggered an update to the store (conversation/messages) takes more time, which results in performance issues https://github.com/chatwoot/chatwoot-mobile-app/issues/504

# Message issues

- Does not have a temporary message available.
    - Implement the same logic used in the dashboard
- Re-rendering issues with a couple of messages
- Try sending more attachments - Previous attachments will start re-render again
- The issue with rendering emails https://github.com/chatwoot/chatwoot-mobile-app/issues/439, https://github.com/chatwoot/chatwoot-mobile-app/issues/475
- Re-rendering profile pic in chat screen header

# Missing features

- Twitter conversation improvements - https://github.com/chatwoot/chatwoot-mobile-app/issues/337
- Outgoing restriction ins Whatsapp and twiiter - https://github.com/chatwoot/chatwoot-mobile-app/issues/424
- Snooze
- Mute → https://github.com/chatwoot/chatwoot-mobile-app/issues/295
- In-app notifications → https://github.com/chatwoot/chatwoot-mobile-app/issues/381
- Team Assignment → https://github.com/chatwoot/chatwoot-mobile-app/issues/323
- Label assignment → https://github.com/chatwoot/chatwoot-mobile-app/issues/331
- Rich content editor → https://github.com/chatwoot/chatwoot-mobile-app/issues/267
- Expandable reply box
- Ability to change password → https://github.com/chatwoot/chatwoot-mobile-app/issues/332
- Previous conversations → https://github.com/chatwoot/chatwoot-mobile-app/issues/333
- Twitter reply features
- Read Indicator
- Support mentions, teams filter?
- Conversation transcript
- Agent profile details → https://github.com/chatwoot/chatwoot-mobile-app/issues/331
- Show more details about the contact
- Show custom attributes https://github.com/chatwoot/chatwoot-mobile-app/issues/428
- Missing inbox name
- Email Headers
- Conversation transcripts
- Redux tests https://github.com/chatwoot/chatwoot-mobile-app/issues/268

# Bugs

- app not signing out even after the token expires
- Find a way to render texts properly - There are lot places where app crashes if the text is empty - https://github.com/chatwoot/chatwoot-mobile-app/issues/501 https://github.com/chatwoot/chatwoot-mobile-app/issues/438
- Improve account switching → https://github.com/chatwoot/chatwoot-mobile-app/issues/500
- Allow only conversation URL in deep linking https://github.com/chatwoot/chatwoot-mobile-app/issues/408

# UX Improvements

- Notification improvements → https://github.com/chatwoot/product/issues/380
- Bloated notifications
- Unable to identify the inbox from which the conversation is created
- Unable to identify the account
- Better conversation resolve UX - https://github.com/chatwoot/chatwoot-mobile-app/issues/477
- Add title and close icon in action sheet view https://github.com/chatwoot/chatwoot-mobile-app/issues/329
- The WhatsApp channel icon is missing ([app.chatwoot.com](http://app.chatwoot.com/))
- Change bot-message colour, add “Sent by: bot”?
- Show “Identity of this user is not verified”
- Show password on the login screen
- Add icons for the conversation context menu
- Expandable reply box
- Add icons for messenger, Instagram
- Add status “All” in the conversation filter
- Add sender avatar in each message
- Add icon for last activity in conversation item
- Support for Subject in Email
- Rating option
- Add an info icon near the menu on the chat screen, it should open the contact details page

# Documentation

- Fastlane release https://github.com/chatwoot/chatwoot-mobile-app/issues/396
- Push notification guide → [https://developers.mattermost.com/contribute/mobile/push-notifications/](https://developers.mattermost.com/contribute/mobile/push-notifications/)

# **Questions**

- Can we share the logic between the mobile app and the dashboard?
- How to track sentry errors?

### PR DESCRIPTION
**Pull request by zaronwald**
# Pull Request Template

## Description 

Implemented the showing and hiding of password on the login screen, by adding a touchable eye image that changes the security of the password field

Fixes #506 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by zaronwald** [comment]
@muhsin-k I created a pull request to add the eye for V2 of the app for the password on the login page

File: ios/Chatwoot.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -605,7 +605,7 @@
 					"-ObjC",
 					"-lc++",
 				);
-				PRODUCT_BUNDLE_IDENTIFIER = com.chatwoot.app;
+				PRODUCT_BUNDLE_IDENTIFIER = com.zach.chatwoot.app;
```
Comment (muhsin-k) [inline ios/Chatwoot.xcodeproj/project.pbxproj:608]: Please revert this.

File: ios/Chatwoot.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -557,8 +557,8 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CLANG_ENABLE_MODULES = YES;
 				CODE_SIGN_ENTITLEMENTS = Chatwoot/Chatwoot.entitlements;
-				CURRENT_PROJECT_VERSION = 325;
-				DEVELOPMENT_TEAM = L7YLMN4634;
+				CURRENT_PROJECT_VERSION = 324;
+				DEVELOPMENT_TEAM = QX4UR6UAR2;
```
Comment (muhsin-k) [inline ios/Chatwoot.xcodeproj/project.pbxproj:561]: Please revert this.

Comment (zaronwald) [reply ios/Chatwoot.xcodeproj/project.pbxproj]: fixed

File: ios/Chatwoot.xcodeproj/project.pbxproj (hunk 1 above)
Comment (zaronwald) [reply ios/Chatwoot.xcodeproj/project.pbxproj]: done

File: src/screens/ConversationDetails/components/ConversationParticipants.js [hunk 3]
```diff
@@ -191,15 +191,19 @@ const ConversationParticipants = ({ conversationId }) => {
                 <View />
               )}
               {isCurrentUserWatching ? (
-                <Text sm color={colors.textGrayLighter}>
+                <Text accessible={true} sm color={colors.textGrayLighter}>
```
Comment (muhsin-k) [inline src/screens/ConversationDetails/components/ConversationParticipants.js:194]: Lets do the all the accessibility changes in a separate PR.

**Review by muhsin-k — COMMENTED** [review]
Thanks for the PR. Could you please fix the conflicts?

**Comment by muhsin-k** [comment]
Closing this since there hasn't been any activity on this for a while. Let's reopen if required.

**Pull request closed without merging**